tests/CrudPanel/CrudPanelColumnsTest.php

```diff
@@ -0,0 +1,163 @@
+
+        $this->assertEquals(2, count($this->crudPanel->columns));
+        $this->assertContains($expectedColumns[0], $this->crudPanel->columns);
+        $this->assertContains($expectedColumns[1], $this->crudPanel->columns);
+    }
+
+    /** @test */
+    public function it_can_add_columns()
+    {
+        $expectedColumns = [
+            [
+                'name'  => 'column',
+                'label' => 'Column',
+            ],
+            [
+                'name' => 'column2',
+                'label' => 'Column2',
+            ],
+        ];
+
+        $this->crudPanel->addColumns($expectedColumns);
+
+        $this->assertEquals(2, count($this->crudPanel->columns));
+        $this->assertContains($expectedColumns[0], $this->crudPanel->columns);
+        $this->assertContains($expectedColumns[1], $this->crudPanel->columns);
+    }
+
+    /** @test */
+    public function it_can_move_column_before()
+    {
+        $expectedColumns = [
+            [
+                'name'  => 'column',
+                'label' => 'Column',
+            ],
+            [
+                'name' => 'column2',
+                'label' => 'Column2',
+            ],
+        ];
+
+        $this->crudPanel->addColumns($expectedColumns);
+        $this->assertEquals(2, count($this->crudPanel->columns));
+        $this->assertContains($expectedColumns[0], $this->crudPanel->columns);
+        $this->assertContains($expectedColumns[1], $this->crudPanel->columns);
+
+        $this->crudPanel->beforeColumn('column');
+        $keys = array_keys($this->crudPanel->columns);
+        $this->assertEquals($expectedColumns[1], $this->crudPanel->columns[$keys[0]]);
+
+        $this->assertEquals(['column2', 'column'], $keys);
+    }
+
+    /** @test */
+    public function it_can_move_column_after()
+    {
+        $expectedColumns = [
+            [
+                'name'  => 'column',
+                'label' => 'Column',
+            ],
+            [
+                'name' => 'column2',
+                'label' => 'Column2',
+            ],
+            [
+                'name' => 'column3',
+                'label' => 'Column3',
+            ],
+        ];
+
+        $this->crudPanel->addColumns($expectedColumns);
+        $this->assertTrue(count($this->crudPanel->columns) == 3);
+        $this->assertContains($expectedColumns[0], $this->crudPanel->columns);
+        $this->assertContains($expectedColumns[1], $this->crudPanel->columns);
+        $this->assertContains($expectedColumns[2], $this->crudPanel->columns);
+
+        $this->crudPanel->afterColumn('column');
+        $keys = array_keys($this->crudPanel->columns);
+
+        $this->assertEquals($expectedColumns[2], $this->crudPanel->columns[$keys[1]]);
+
+        $this->assertEquals(['column', 'column3', 'column2'], $keys);
+    }
+
+    /** @test */
+    public function it_can_remove_column_by_name()
+    {
+        $this->crudPanel->addColumns(['column1', 'column2', 'column3']);
```

Comment:
There should be something to indicate that this isn't actually passing? Skip test or something?